src/main/java/uk/ac/sanger/sccp/stan/service/history/HistoryServiceImp.java

```diff
@@ -130,7 +135,16 @@ public History getHistoryForWorkNumber(String workNumber) {
         }
         List<Sample> samples = referencedSamples(entries, allLabware);
         entries.sort(Comparator.comparing(HistoryEntry::getTime));
-        return new History(entries, samples, allLabware);
+        List<SamplePositionResult> samplePositionResults = getAllSamplePositionResults(allLabware);
+        return new History(entries, samples, allLabware, samplePositionResults);
+    }
+
+    private List<SamplePositionResult> getAllSamplePositionResults(List<Labware> labwareList) {
+        List<SamplePositionResult> samplePositionResultList = new ArrayList<>();
+        for(Labware labware : labwareList) {
+           samplePositionResultList.addAll(slotRegionService.loadSamplePositionResultsForLabware(labware.getBarcode()));
```

Comment:
Rather than calling `loadSamplePositionResultsForLabware(barcode)` repeatedly, could add a method to SlotRegionService, i.e.
```

    @Override
    public List<SamplePositionResult> loadSamplePositionsResultsForLabware(Collection<Labware> labware) {
        Map<Integer, Slot> slotIdMap = labware.stream()
                .flatMap(lw -> lw.getSlots().stream())
                .collect(BasicUtils.inMap(Slot::getId));
        List<SamplePosition> sps = samplePositionRepo.findAllBySlotIdIn(slotIdMap.keySet());
        if (sps.isEmpty()) {
            return List.of();
        }
        return sps.stream()
                .map(sp -> toSamplePositionResult(sp, slotIdMap))
                .collect(toList());
    }
```